modules/phase_field/src/kernels/ACInterfaceBetaPenalty.C

```diff
@@ -0,0 +1,85 @@
+// Considers cleavage plane anisotropy in the crack propagation
+
+#include "ACInterfaceBetaPenalty.h"
+
+registerMooseObject("PhaseFieldApp", ACInterfaceBetaPenalty);
+
+template <>
+InputParameters
+validParams<ACInterfaceBetaPenalty>()
+{
+  InputParameters params = validParams<ACInterface>();
+  params.addClassDescription("Gradient energy Allen-Cahn Kernel where crack propagation along weak "
+                             "cleavage plane is preferred");
+  params.addRequiredParam<Real>(
+      "beta_penalty",
+      "penalty to penalize fracture on planes not normal to one cleavage plane "
+      "normal which is normal to weak cleavage plane. Setting beta=0 results "
+      "in isotropic damage.");
+  params.addRequiredParam<std::vector<Real>>("cleavage_plane_normal",
+                                             "Normal to the weak cleavage plane");
+  return params;
+}
+
+ACInterfaceBetaPenalty::ACInterfaceBetaPenalty(const InputParameters & parameters)
+  : ACInterface(parameters),
+    _beta_penalty(getParam<Real>("beta_penalty")),
+    _cleavage_plane_normal(getParam<std::vector<Real>>("cleavage_plane_normal"))
+{
+}
+
+Real
+ACInterfaceBetaPenalty::betaNablaPsi()
+{
+  Real ngradc;
+  ngradc = _grad_u[_qp](0) * _cleavage_plane_normal[0] +
```

Comment:
can be simplified as _grad_u[_qp] * _clevarge_plane_normal.

Same for the ngradpsi below.

Reply:
@jiangwen84 
_grad_u[_qp] * _clevarge_plane_normal  did not work.
It gives me error:
'const value_type {aka const libMesh::VectorValue<double>}' is not derived from 'const std::complex<_Tp>'

The declaration in the header file is   const std::vector<Real> _cleavage_plane_normal;
Does this multiplication needs any other type ?